### PR TITLE
Sync v2-7-stable with v2-7-test to release 2.7.0

### PR DESCRIPTION
**Pull request by ephraimbuddy**
Time for `2.7.0rc2`

### CODE REVIEW

**Review by hussein-awala — APPROVED** [review]
Looks good, all the fixes we discussed recently are included!

**Review by phanikumv — APPROVED** [review]

**Review by potiuk — APPROVED** [review]
LGTM. There are a few "dev" related small fixes that I merged yesterday (that were about to complete the "eager upgrade" problem resolution  - that will need to be likely cherry-picked. But other than that all good.

**Review by pierrejeambrun — APPROVED** [review]
Nice

File: RELEASE_NOTES.rst [hunk 1]
```diff
@@ -37,6 +37,13 @@ Old Graph View is removed (#32958)
 """"""""""""""""""""""""""""""""""
 The old Graph View is removed. The new Graph View is the default view now.
 
+The trigger UI form is skipped in web UI with 2.7.0 if no parameters are defined in a DAG (#33351)
+""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
```
Comment (jedcunningham) [inline RELEASE_NOTES.rst]:
```suggestion
The trigger UI form is skipped in web UI if no parameters are defined in a DAG (#33351)
"""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
```

nit

File: RELEASE_NOTES.rst [hunk 2]
```diff
@@ -332,6 +340,8 @@ Misc/Internal
 
 Docs only changes
 """""""""""""""""
+- Fix links in security docs (#33329)
```
Comment (jedcunningham) [inline RELEASE_NOTES.rst]:
```suggestion
```

Don't need this one in the release notes.

**Review by jedcunningham — APPROVED** [review]

**Review by eladkal — APPROVED** [review]
We need also to wait for a fix for labels issue

https://apache-airflow.slack.com/archives/CCPRP7943/p1692086197633229

**Pull request merged**